src/test/resources/com/puppycrawl/tools/checkstyle/checks/modifier/classmemberimpliedmodifier/InputClassMemberImpliedModifierOnClass.java

```diff
@@ -0,0 +1,69 @@
+package com.puppycrawl.tools.checkstyle.checks.modifier.classmemberimpliedmodifier;
+
+public class InputClassMemberImpliedModifierOnClass {
```

Comment:
Done

Reply:
this file is used in several test methods and not only with default configurations.
Whole idea to put config in some form to javadoc to read config and clearly see violation location in file (without scrolling back to UT and verifying line numbers).
Please make a copy of this Input file as much as you need, place config in javadoc and reflect expected violations from UT method in input as trailing comment. 

... One day we will make automatic validation of this pattern.

Reply:
FWIW, I think duplicating the test files (many times) simply to get Javadoc/violation comments is a terrible idea. It will result in multiple copies of the file which will be significantly more confusing to readers/maintainers. The violation text I added refers to the things that the check normally treats as violations - let the each individual test switch them on and off (as any normal test code would do).

Reply:
@rnveach , please step in for second option.
In case i am losing point

Reply:
> multiple copies of the file which will be significantly more confusing to readers/maintainers

I disagree on this point as file will contain configuration that was used and clearly state what code causes a violation with that specific config. I wouldn't find this confusing, it would just add many more files to keep as each config setup would require a new input.

We usually don't `maintain` these files per say, because they are snapshot in time on what the user gave us in issue. Adding on to the end of the file is ok as long as previous line/column violations don't change so git history will showcase how code change affected violations just like our regression.

It is slightly more work to start these files up, but copying pasting the exact same contents and changing comments is basically all it amounts to. Same file with different configurations allows us to show how check can be configured and how it will behave differently.

Reply:
Done